### DISCUSSION TITLE
Disable anchors selectively for chunks

### DISCUSSION THREAD

**Discussion by royfrancis** [quarto-dev/quarto-cli#10274] [Q&A] [answered]
### Description

A code chunk with a label creates a clickable anchor. This is usually fine and nice to have.

![Screenshot 2024-07-11 at 18 33 42](https://github.com/quarto-dev/quarto-cli/assets/5083500/ff53e0f5-b5ac-4784-bb16-f974ab7af9de)

But sometimes, with interactive graphics, it can conflict with existing elements.

![Screenshot 2024-07-11 at 18 34 15](https://github.com/quarto-dev/quarto-cli/assets/5083500/1c55c40f-bce7-4e1c-8139-3419a75a802a)

`anchor-sections: false` disables anchors for the whole document. Is it possible to disable anchors selectively by chunk? Or enable anchors for headings, but disable for figures?

An example document

<details>
<summary>Click here</summary>

````
---
title: Anchor conflict
format: html
---

## ggplot

```{r}
#| label: fig-ggplot
#| fig-cap: A ggplot figure

library(ggplot2)
data <- data.frame(x = c(1, 2, 3, 4, 5), y = c(10, 11, 12, 13, 14))
p <- ggplot(data = data,aes(x = x,y = y))+
    geom_point()

p+labs(title="A ggplot figure")
```

## ggiraph

```{r}
#| label: fig-ggiraph
#| fig-cap: A ggiraph figure

library(ggiraph)
p1 <- p+labs(title="A ggiraph figure")
girafe(ggobj = p1) |>
    girafe_options(opts_toolbar(saveaspng = TRUE))
```

## plotly

```{r}
#| label: fig-plotly
#| fig-cap: A plotly figure

library(plotly)
plot_ly(data, x = ~x, y = ~y, type = 'scatter', mode = 'markers') |>
layout(title = list(text = "A plotly figure", x = 0.1))
```
````

</details>

`quarto 1.5.54`

**Comment by cderv** [comment]
There is no way for now to opt out anchor on specific element. It is all of nothing through `anchor-sections` configuration. 

There is a trick though using internal logic we have (undocumented but can be a workaround for now for quarto 1.5)

**To be clear: No guarantee to keep this as it is in future versions**

````lua
Div = function(div)
  -- get div that have a no-anchor class
  if div.classes:includes('no-anchor') then
    -- Remove this class we don't need in output
    div.classes = div.classes:filter(function(class)
      return class ~= 'no-anchor'
    end)
    return div:walk({
      Div = function(div)
        -- check all div inside and add no-anchor on figure with an id
        if div.classes:includes('quarto-figure') and #div.identifier > 0 then
          div.classes:insert('no-anchor')
        end
        return div
      end
    })
  end
end

````

and add this to your YAML header
````yaml
filters:
  - at: post-render
    path: no-anchor.lua
````

and adding in qmd

````markdown
## ggiraph

```{r}
#| label: fig-ggiraph
#| fig-cap: A ggiraph figure
#| classes: no-anchor

library(ggiraph)
p1 <- p+labs(title="A ggiraph figure")
girafe(ggobj = p1) |>
    girafe_options(opts_toolbar(saveaspng = TRUE))
```
````

Probably in the future, we'll need to add a way for this `no-anchor` opt-out to be an exposed feature. It can be set easily on headings for section I believe, but not for computed figure.

Hope this helps. And thanks for the idea.

**Reply by cderv** [reply, marked as answer]
- https://github.com/quarto-dev/quarto-cli/issues/10275

**Reply by royfrancis** [reply]
Perhaps a simple solution to fix the overlap would be just to make sure that the chunk anchors are not positioned over the output. ie; It is positioned to the right for example, with some padding/margin etc? I was thinking similar to how the heading anchors are positioned to the right of the headings. Easy for me to say but probably creates other issues 😄

**Reply by cderv** [reply]
> but probably creates other issues

That could indeed cause other problem. It would mean putting it outside of the body I guess, a little in margin, but we could indeed think of something to avoid this type of conflict
